client_internals_it_test.go

```diff
@@ -542,3 +543,176 @@ func ensureClient(config hz.Config) *hz.Client {
+		name          string
+		muStatus      MutualAuthenticationStatus
+		ca, cert, key string
+		wantErr       bool
+	}{
+		{
+			name:     "testMA_RequiredClientAndServerAuthenticated",
+			muStatus: REQUIRED,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "testdata/OpenSSL/client/client.crt",
+			key:      "testdata/OpenSSL/client/client.key",
+			wantErr:  false,
+		},
+		{
+			name:     "testMA_RequiredServerNotAuthenticated",
+			muStatus: REQUIRED,
+			ca:       "testdata/unknown-server-cert.pem",
+			cert:     "testdata/OpenSSL/client/client.crt",
+			key:      "testdata/OpenSSL/client/client.key",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_RequiredClientNotAuthenticated",
+			muStatus: REQUIRED,
+			ca:       "testdata/OpenSSL/rootCa.crt",
+			cert:     "testdata/unknown-client-cert.pem",
+			key:      "testdata/unknown-client-key.pem",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_RequiredClientAndServerNotAuthenticated",
+			muStatus: REQUIRED,
+			ca:       "testdata/unknown-server-cert.pem",
+			cert:     "testdata/unknown-client-cert.pem",
+			key:      "testdata/unknown-client-key.pem",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_OptionalClientAndServerAuthenticated",
+			muStatus: OPTIONAL,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "testdata/OpenSSL/client/client.crt",
+			key:      "testdata/OpenSSL/client/client.key",
+			wantErr:  false,
+		},
+		{
+			name:     "testMA_OptionalServerNotAuthenticated",
+			muStatus: OPTIONAL,
+			ca:       "testdata/unknown-server-cert.pem",
+			cert:     "testdata/OpenSSL/client/client.crt",
+			key:      "testdata/OpenSSL/client/client.key",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_OptionalClientNotAuthenticated",
+			muStatus: OPTIONAL,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "testdata/unknown-client-cert.pem",
+			key:      "testdata/unknown-client-key.pem",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_OptionalClientAndServerNotAuthenticated",
+			muStatus: OPTIONAL,
+			ca:       "testdata/unknown-server-cert.pem",
+			cert:     "testdata/unknown-client-cert.pem",
+			key:      "testdata/unknown-client-key.pem",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_RequiredWithNoCertFile",
+			muStatus: REQUIRED,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "",
+			key:      "",
+			wantErr:  true,
+		},
+		{
+			name:     "testMA_OptionalWithNoCertFile",
+			muStatus: OPTIONAL,
+			ca:       "testdata/OpenSSL/rootCA.crt",
+			cert:     "",
+			key:      "",
+			wantErr:  false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			skip.IfNot(t, "oss")
+			if tc.wantErr {
+				require.Error(t, sslTestUtil(t, tc.muStatus, tc.ca, tc.cert, tc.key))
+			} else {
+				require.NoError(t, sslTestUtil(t, tc.muStatus, tc.ca, tc.cert, tc.key))
+			}
+		})
+	}
+}
+
+func sslTestUtil(t *testing.T, mu MutualAuthenticationStatus, ca, cert, key string) error {
```

Comment:
How about `status` instead of `mu`?

Reply:
I think no need to pass `t` since it is being used only for `t,Name()`. You can pass the cluster name instead.

Reply:
Fixed, [c6481ec](https://github.com/hazelcast/hazelcast-go-client/pull/859/commits/c6481ec04328cf044ab3bec8e31bd712bd663bd2).